### ISSUE TITLE
Collections: Add counts to search results

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#1745]
### Descriptive summary

The count of works and collections should be part of the search results displayed with collections in search results.  This is part of the remaining work from issue #1670 - Calculate the number of items in a collection.

### Rationale

This provides the user additional information and makes collections more visually distinct in search results.

### Expected behavior

The UI for search results needs to be updated to display the work count and subcollection count for collections in search results.  The display should match the mockup (below).

Use the following to get the number of works...
```
<%=collection_presenter.total_viewable_works%>
```
and this for the number of subcollections...
```
<%=collection_presenter.total_viewable_collections%>
```

### Steps to reproduce the behavior

1. Create a collection type that allows discovery
1. Create a collection of that type with at least 3 works and a subcollection (of the same type) as members.
1. Search for the name in the search bar on the home page.

Verify that the collection is in the search results and that the count of works and count of subcollections is correct.

### Mockups

![image](https://user-images.githubusercontent.com/6855473/30441935-e32391ba-9948-11e7-8b53-d3cce0c05e58.png)

### Related work

#1670 Calculate the number of items in a collection
#1739 Collection presenters have counts of viewable members by type and total

**Comment by adamjarling** [comment]
@elrayle I can't find how to add a collection as a sub-collection from the UI.  Is this path available in `collections-sprint` at the moment?   Looks like the relationships tab, and nesting options both are still inactive... Maybe I'm missing something.

**Comment by adamjarling** [comment]
@elrayle Regarding the Slack comment about not over-writing other pieces of the application which use the partials I've updated in this branch... I did a bunch of home page, app searches for Collections and terms (granted with only app data added in the Issue description above), and I think I found how to make this an update which plays better with Bootstrap's usage of rows/cols for Collections and Works which are returned.  

Is there a way I could do a hand-off at this point?  I think the remaining work would be:

- Wire up the Collections count total, and the Works count total in `app/views/catalog/_index_list_default.html.erb`.  I tried plugging in the sample code you supplied above to output the counts, however I'm not sure there `collection_presenter` comes from, and it didn't work.
- Determine whether a search result should display the new Collection and Work counts, and add  display logic to `app/views/catalog/_index_list_default.html.erb`
- Verify tests pass.
